### PR TITLE
[bugfix] Correctly import XQuery library modules from JAR files

### PR DESCRIPTION
**Pull request by adamretter**

### CODE REVIEW

File: src/org/exist/source/SourceFactory.java [hunk 1]
```diff
@@ -58,19 +64,26 @@
...
+        } else if(contextPath != null && contextPath.startsWith(ClassLoaderSource.PROTOCOL)) {
+            // Pretend it is a file on the local system so we can resolve it easily with URL() class.
+            final String conPathNoProtocol = contextPath.replace(ClassLoaderSource.PROTOCOL, "file://");
+            String resolvedURL = new URL(new URL(conPathNoProtocol), location).toString();
+            resolvedURL = resolvedURL.replaceFirst("file://", ClassLoaderSource.PROTOCOL);
+            source = new ClassLoaderSource(resolvedURL);
+        }
         /* file:// or location without scheme is assumed to be a file */
-        if(location.startsWith("file:") || location.indexOf(':') == Constants.STRING_NOT_FOUND)
+        else if(location.startsWith("file:") || location.indexOf(':') == Constants.STRING_NOT_FOUND)
```
Comment (dizzzz) [inline src/org/exist/source/SourceFactory.java]: Constants.STRING_NOT_FOUND ... can be replaced with a Java6+ operation on the string object?

File: src/org/exist/source/SourceFactory.java [hunk 2]
```diff
@@ -58,19 +64,26 @@
...
         {
             location = location.replaceAll("^(file:)?/*(.*)$", "$2");
 
-            final File f = new File(contextPath + File.separatorChar + location);
-            if(f.canRead())
-            {
-
+            final File f = new File(contextPath, location);
```
Comment (dizzzz) [inline src/org/exist/source/SourceFactory.java]: no Java7+ Path ?

**Comment by dizzzz** [comment]
my remarks are not critical.....

**Comment by ljo** [comment]
Yes, looks good and comments valid.

File: src/org/exist/source/SourceFactory.java (hunk 1 above)
Comment (adamretter) [inline src/org/exist/source/SourceFactory.java]: This was existing code which I didn't modify.

File: src/org/exist/source/SourceFactory.java (hunk 2 above)
Comment (adamretter) [inline src/org/exist/source/SourceFactory.java]: Likewise this was existing code that I didn't rewrite.

**Comment by adamretter** [comment]
@dizzzz @ljo I just appended a commit to this that rewrites a bunch more of the old code to `java.nio.file`, this at least addresses the concerns that you raised. Hopefully all is good now?

**Pull request merged**